src/rust/src/declarative_asn1/decode.rs

```diff
@@ -57,6 +59,50 @@ fn decode_printable_string<'a>(
     Ok(pyo3::Bound::new(py, PrintableString { inner })?)
 }
 
+fn decode_utc_time<'a>(
+    py: pyo3::Python<'a>,
+    parser: &mut Parser<'a>,
+) -> ParseResult<pyo3::Bound<'a, UtcTime>> {
+    let value = parser.read_element::<asn1::UtcTime>()?;
+    let dt = value.as_datetime();
+
+    let inner = pyo3::types::PyDateTime::new(
+        py,
+        dt.year().into(),
+        dt.month(),
+        dt.day(),
+        dt.hour(),
+        dt.minute(),
+        dt.second(),
+        0,
+        Some(&pyo3::types::PyTzInfo::utc(py)?.to_owned()),
+    )?
+    .unbind();
```

Comment:
I think we have a function that does this -- `datetime_to_py_utc`. Your implementation is better than that one, so assuming it works on all python versions, can you replace that impl with this one and call it here?

Reply:
Actually -- if you can specifically do a standalone PR that changes those functions to use pyo3's types that'd be best.

Reply:
PR open here: https://github.com/pyca/cryptography/pull/13514